crates/tarball/src/lib.rs

```diff
@@ -115,7 +115,7 @@ fn verify_checksum(data: &[u8], integrity: Integrity) -> Result<ssri::Algorithm,
 #[instrument(skip(cache), fields(cache_len = cache.len()))]
 pub async fn download_tarball_to_store(
     cache: &Cache,
-    store_dir: &Path,
+    store_dir: Arc<PathBuf>,
```

Comment:
`Arc<Path>` is better because less pointer indirection.

If you worry about creating `Arc<Path>` from `PathBuf`, just use `.into()`.